src/sedml/SedAbstractCurve.cpp

```diff
@@ -68,6 +67,7 @@ SedAbstractCurve::SedAbstractCurve(unsigned int level, unsigned int version)
   , mElementName("abstractCurve")
 {
   setSedNamespacesAndOwn(new SedNamespaces(level, version));
+  mNameAllowedPreV4 = true;
```

Comment:
should we not have id allowed prev4 too?

Reply:
Ah, this seems to be a UML vs. the text thing.  The UML diagram doesn't have it, and the text doesn't list it, but it's in the table.  (This was true in l1v2, too).

For me, if the spec was ambiguous, I'm OK with allowing the ID in those level/versions, especially since we added it later.  I'll make the fix.

Reply:
...except now that I look at the other tables, every single table in the spec that lists 'id' also lists 'name', and visa versa.  So while the UML diagrams indicated that some classes only had 'name' and some only had 'id', the tables always listed them together.

Do we just say that everything had both 'id' and 'name' that had either?  I suppose this might be a question for the editors.  My inclination is to allow both.

Reply:
OK, here's a breakdown of the only-id-or-name classes:

* Range:  only 'id' in the UML; no table with allowed attributes, so nothing ever mentions 'name' being allowed.
* AbstractCurve:  only 'name' in the UML; both 'id' and 'name' in Table 2.28
* DataSet:  only 'name' in the UML; both 'id' and 'name' in Table 2.30
* Surface:  only 'name' in the UML; both 'id' and 'name' in Table 2.29  (Also, inherits from Curve.)

I've filed https://github.com/SED-ML/sed-ml/issues/139 to ask what other editors think.

Reply:
well, since libSEDML always allowed to set an id attribute on these elements, i'm hesitant to just remove that for now, so i would want it to be set / read / written from now on too (if the user would set it)